src/proxy-animation.js

```diff
@@ -8,120 +8,278 @@ import {
 const nativeElementAnimate = window.Element.prototype.animate;
 const nativeAnimation = window.Animation;
 
-/**
- * Decides whether to use native Element.prototype.animate function in regular fashion or pass it to our polyfill
- *  so its current time is driven by scroll event
- * @param keyframes {Object} array of keyframe objects
- * @param options {Object} WAAPI options object
- * @returns {Function}
- */
+class PromiseWrapper {
+  constructor(details) {
+    this.state = 'pending';
+    this.details = details;
+    this.promise = new Promise((resolve, reject) => {
+      this.nativeResolve = resolve;
+      this.nativeReject = reject;
+    });
+  }
+  resolve() {
+    this.state = 'resolved';
+    this.nativeResolve(this.details.proxy);
+  }
+  reject() {
+    this.state = 'rejected';
+    // Do not report unhandled promise rejections.
```

Comment:
This is to prevent the unhandled exceptions warnings which having been deprecated and will cause node to fail soon.  Cancelling a pending animation where we don't have an explicit catch will generate an unhandled exception otherwise.  I have not been able to find a quick spec link stating explicitly that unhandled rejections of the ready/finished promises should be marked as handled; however, it is tested via the harness (these were showing up as harness errors).  See:
https://source.chromium.org/chromium/chromium/src/+/master:third_party/blink/renderer/core/animation/animation.cc;l=1690?q=blink%20%2Fanimation.cc&ss=chromium%2Fchromium%2Fsrc

Can try again to find a spec link.

Reply:
To add a specific example:
animation.play();
animation.cancel();

would trigger an unhandled exception on the ready promise.  We do not require the user to specifically add animation.ready.then|reject|catch to handle the eception.  The empty catch marks these exceptions as handled. With this change "Harness Tests Successfuly Completed" goes from 27/35 to 31/35.  Doesn't affect the number of passing tests but the number of passing test harnesses.

Reply:
Got it, sounds good. I misunderstood the intention. That makes perfect sense that not handling the rejection should not throw an error.